tracer/src/Datadog.Trace/Agent/MessagePack/SpanMessagePackFormatter.cs

```diff
@@ -189,9 +197,57 @@ private int Serialize(ref byte[] bytes, int offset, in SpanModel spanModel)
             offset += WriteTags(ref bytes, offset, in spanModel, tagProcessors);
             offset += WriteMetrics(ref bytes, offset, in spanModel, tagProcessors);
 
+            if (metaStructValues?.Count > 0)
+            {
+                WriteMetaStruct(ref bytes, ref offset, metaStructValues);
+            }
+
             return offset - originalOffset;
         }
 
+        private void WriteMetaStruct(ref byte[] bytes, ref int offset, IReadOnlyDictionary<string, object> metaStruct)
+        {
+            var data = new Dictionary<string, byte[]>(metaStruct.Count);
+
+            // Since ConcurrentDictionary does not guarantee the order of the keys, we need to order them
+            // to make serialization deterministic.
```

Comment:
We should think if being deterministic worth the hassle.

Also we already have all the mechanism for Tags (string) and Metrics (doubles) why not create MetaStruct (objects or byte[]) in the same way? why are we involving all this concurrentdictionary and dictionary stuff?

Reply:
Done!